backend/scripts/github_collector.py

```
@@ -284,26 +428,289 @@ def collect_all(self):
+                 pass
+
+        # Check if we have cached branches in ChromaDB (File System Check)
+        if check_cache or local_only:
+            cache_check = self.check_chromadb_exists(self.github_repo)
+            if cache_check.get("exists") and cache_check.get("count", 0) > 0:
+                # Extract branch names from existing ChromaDB directories
+                cached_branches = []
+                for db_path in cache_check.get("databases", []):
+                    # Extract branch name from path like "./chroma_db_owner_repo_branchname"
+                    # Remove the prefix to get the sanitized branch name
+                    prefix = os.path.join(base_chroma_path, f"chroma_db_{self.repo_safe_name}_")
```

Ruff (0.14.8)

Undefined name `base_chroma_path`

(F821)